### PR TITLE
Project 5: Refactor code

### PR DESCRIPTION
**Pull request by Nelitaa**
### Project requirements

At this point your codebase is pretty big. The `main.rb` file may be especially large at this point. Let's try to clean things up!

- Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Pay special attention to the "Single-responsibility Principle".
    - Make sure that the role of `main.rb` is only to start your code and ask the user for the initial choice of options (or to quit).
    - Extract some pieces of logic into separate class(es).

### CODE REVIEW

**Review by Gambit142 — COMMENTED** [review]
Hi @Nelitaa 👋, 

While you made a great effort in this project, I can't go ahead and review your code.

## Invalid Code Review Request

- I understand that this is a pair-programming activity, but you are the only one that worked on this project. Kindly collaborate with your partner for this milestone 👍.


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person_manager.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    person = gets.chomp.to_i
+    case person
+    when 1
+      add_student
+    when 2
+      create_teacher
+    else
+      puts "Invalid Option!\n "
+    end
+  end
```
Comment (rloterh) [inline person_manager.rb]: - [ ] The `add_person` method does not adhere to the single responsibility rule. Kindly ensure that *all* requests for input and input validation are placed in the `main.rb` file and/or classes/methods whose responsibilities are to request or validate input.` I urge you to kindly read further on the same via [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)

File: person_manager.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, age, name: name)
+    @people.push(teacher)
+    puts "\nPerson created successfully.\n "
+  end
```
Comment (rloterh) [inline person_manager.rb]: - [ ] Similarly, `add_student` and the `create_teacher` methods do not adhere to the single responsibility rule. Kindly ensure that *all* requests for input and input validation are placed in the `main.rb` file and/or classes/methods whose responsibilities are to request or validate input.`

File: rental_manager.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  def list_rentals_for_person_id
+    print 'ID of person: '
+    id = gets.chomp.to_i
+    puts 'Rentals:'
+    @rentals.each do |rental|
+      puts "Date: #{rental.date}, Book: '#{rental.book.title}' by #{rental.book.author}" if rental.person.id == id
+    end
+    puts "End of list.\n "
+  end
```
Comment (rloterh) [inline rental_manager.rb]: - [ ] Again, `add_rental` and the `list_rentals_for_person_id` methods do not adhere to the single responsibility rule. Kindly ensure that *all* requests for input and input validation are placed in the `main.rb` file and/or classes/methods whose responsibilities are to request or validate input.`

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Nela & Youssef, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦🚦   🏃 🏃‍♂️  

## Highlights 👍  
- Good job refactoring to improve your code so far! 
- Nice job marking your object with _accessor attributes_ before initializing them in a constructor!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @TEAM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Good implementations ✔️ 
- No linter errors ✔️ 
- Main file working well ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**